riot/tft/summoner.go

```diff
@@ -0,0 +1,63 @@
+package tft
+
+import (
+	"fmt"
+	"github.com/KnutZuidema/golio/internal"
+	log "github.com/sirupsen/logrus"
+)
+
+// SummonerClient provides methods for summoner endpoints of the League of Legends TFT API.
+type SummonerClient struct {
+	c *internal.Client
+}
+
+// GetSummonerByAccountID returns a summoner by account ID
+func (sc *SummonerClient) GetSummonerByAccountID(encryptedAccountID string) (*Summoner, error) {
+	logger := sc.logger().WithField("method", "GetSummonerByAccount")
+	url := fmt.Sprintf(endpointSummonerByAccount, encryptedAccountID)
+	var out *Summoner
+	if err := sc.c.GetInto(url, &out); err != nil {
+		logger.Debug(err)
+		return nil, err
+	}
+	return out, nil
+}
+
+// GetSummonerByPUUID returns a summoner by PUUID
+func (sc *SummonerClient) GetSummonerByPUUID(puuid string) (*Summoner, error) {
+	logger := sc.logger().WithField("method", "GetSummonerByPUUID")
+	url := fmt.Sprintf(endpointSummonerByPUUID, puuid)
+	var out *Summoner
+	if err := sc.c.GetInto(url, &out); err != nil {
+		logger.Debug(err)
+		return nil, err
+	}
+	return out, nil
+}
+
+// GetSummonerByMe returns a summoner by access token
+func (sc *SummonerClient) GetSummonerByMe(authorization string) (*Summoner, error) {
+	logger := sc.logger().WithField("method", "GetSummonerByMe")
+	var out *Summoner
+	if err := sc.c.GetInto(endpointSummonerByMe, &out, internal.WithHeader("Authorization", authorization)); err != nil {
```

Comment:
I don't understand the need for this `WithHeader` option. The api key is already available to the client, why do we need additional authorization? Is it different from the API key?

Reply:
This is the API: https://developer.riotgames.com/apis#tft-summoner-v1/GET_getByAccessToken

I had a look at it, and it seem a bit confusing. So golio passes the header `X-Riot-Token` as a header for authenticating the application. But this API requires another token called `Authorization`, it's not specified what this authorization is. But looking at the menu subtitle may give a hint of what it is. The title say `Teamfight tactics RSO`, so the Authorization header that is optional in this API may be the `Riot Sign on`-token. What do you think @KnutZuidema